File: frontend/src/services/milestoneService.js
import api from './api';

export const milestoneService = {
  async getMilestonesByGoal(goalId) {
    const response = await api.get(`/goals/${goalId}/milestones`);
    return response.data;
  },

  async getMilestoneById(id) {
    const response = await api.get(`/milestones/${id}`);
    return response.data;
  },

  async createMilestone(goalId, payload) {
    const response = await api.post(`/goals/${goalId}/milestones`, payload);
    return response.data;
  },

  async updateMilestone(id, payload) {
    const response = await api.put(`/milestones/${id}`, payload);
    return response.data;
  },

  async deleteMilestone(id) {
    const response = await api.delete(`/milestones/${id}`);
    return response.data;
  },
};

export default milestoneService;

File: backend/src/main/java/com/lifeos/milestone/service/MilestoneService.java
package com.lifeos.milestone.service;

import com.lifeos.milestone.dto.CreateMilestoneRequest;
import com.lifeos.milestone.dto.MilestoneResponse;
import com.lifeos.milestone.dto.UpdateMilestoneRequest;

import java.util.List;

public interface MilestoneService {

    MilestoneResponse createMilestone(Long userId, Long goalId, CreateMilestoneRequest request);

    List<MilestoneResponse> getMilestonesByGoal(Long userId, Long goalId);

    MilestoneResponse getMilestoneById(Long userId, Long milestoneId);

    MilestoneResponse updateMilestone(Long userId, Long milestoneId, UpdateMilestoneRequest request);

    void deleteMilestone(Long userId, Long milestoneId);
}
